middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -168,6 +168,266 @@ let find_simple acc env (simple : IR.simple) =
 let find_simples acc env ids =
   List.fold_left_map (fun acc id -> find_simple acc env id) acc ids
 
+module Inlining = struct
+  type inlinable_result =
+    | Not_inlinable
+    | Inlinable of Flambda.Code.t
+
+  let threshold () =
+    let inline_threshold =
+      Clflags.Float_arg_helper.get ~key:0 !Clflags.inline_threshold
```

Comment:
Do you explicitely want to use Flambda1's inline threshold?
If you want to reuse the same flags as the Flambda2's inliner then I think you can remove this function and instead on line 1184 of `closure_conversion_aux.ml` replace `let inline_threshold = Inlining.threshold () in` by:
```
let inline_threshold = 
  Code.inlining_arguments code
  |> Inlining_arguments.small_function_size
  |> Code_size.of_int
in
```

Reply:
It's relatively important to handle the `-inline` flag more or less correctly. I don't think `Inlining_arguments.small_function_size` takes it into account at the moment; so unless this gets fixed it's better to use `Clflags.inline_threshold` directly.

Reply:
I know we tried to make `-inline` work with flambda 1, but it made things quite confusing internally and the result was unsatisfying. Do we really want to do that again, or do we want to just switch people to using `-Ox` instead and have `-inline` mean nothing?

Reply:
Oh sorry, I've made an error, in that case you should use `Inlining_arguments.threshold` directly. It's the same as the one in Flambda1 but it's an int already rescaled by the factor 10.
 One advantage of using `Inlining_arguments` is that they will always be "correct" even if the closure was imported from an external module (which doesn't seem to be the case right now during closure conversion but it's in tune with what the simplifier is using).

Reply:
I was expecting we would have `-inline` do nothing for Flambda 2.

Reply:
I'm in favour of making `-inline` work with Flambda2, even if it's not very satisfying.

For the particular case of this PR, the code is only executed when `-Oclassic` was passed, and there is currently no way to tweak the Flambda 2 inlining parameters except for the `-Ox` flags. So using `Clflags.inline_threshold` allows us to test this pass with varying levels of inlining, and more importantly allows us to match more closely the behaviour of Closure.

We'll try to merge back as much of this code as possible with the main inlining code once we're satisfied that the code is correct, and at that point we may go back to the Flambda 2 inlining arguments, but for now it's far more convenient for us to rely on `Clflags.inline_threshold`.